src/app/review/StudyProgress.js

```diff
@@ -3,21 +3,19 @@ import cx from "classnames";
 
 const StudyProgress = ({ index, items, pageSize, pageStart, pageEnd, isFinished }) => {
   return (
-    <div className="ml-auto mb-2 mx-2">
-      <div className="d-flex align-items-center">
-        {items.slice(pageStart, pageEnd).map((el, key) => (
-          <div
-            key={key}
-            className={cx("border border-dark rounded-circle border-width-2 ml-1", {
-              "bg-dark": isFinished || key < index % pageSize,
-            })}
-            style={{
-              width: "10px",
-              height: "10px",
-            }}
-          />
-        ))}
-      </div>
+    <div className="d-flex align-items-end">
+      {items.slice(pageStart, pageEnd).map((el, key) => (
+        <div
+          key={key}
+          className={cx("border border-dark rounded-circle border-width-2 ml-1", {
```

Comment:
First time seeing this sort of syntax with react, what is its purpose?

Reply:
It's a library called [classnames](https://www.npmjs.com/package/classnames) that supports conditionally adding class names.

Reply:
Here it is conditionally including the `bg-dark` class name on the next line.